buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt

```diff
@@ -63,17 +66,33 @@ open class BuildScanPlugin : Plugin<Project> {
 
         if (isCiServer && !isTravis) {
             extractAllReportsFromCI()
+            monitorUnexpectedCacheMisses()
         }
 
         extractCheckstyleAndCodenarcData()
         extractBuildCacheData()
     }
 
+    private
+    fun Project.monitorUnexpectedCacheMisses() {
+        gradle.taskGraph.afterTask {
+            if (!state.skipped && !currentBuildContainsCompileAll() && this.isMonitored()) {
```

Comment:
It would be great if we could check that the cache hit is actually from the seed build. But I guess that is much harder.

Reply:
There's currently no kind of “web service” in Gradle Enterprise to dereference a (origin) build invocation ID to a scan. We could add it though to support this effort, as any development in our understanding of tooling for build caching usage is worthy.